### PR TITLE
Adding decommission / removal procedure (cloudscale.ch)

### PR DESCRIPTION
**Pull request by nunojusto**

### CODE REVIEW

**Comment by tobru** [comment]
Could you move that into it's own page please?

**Comment by nunojusto** [comment]
> Could you move that into it's own page please?

Sure

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+# configure minio client to use the bucket
+mc config host add \
+  "${CLUSTER_ID}" "https://objects.${REGION}.cloudscale.ch" \
+  $(echo $response | jq -r '.keys[0].access_key') \
+  $(echo $response | jq -r '.keys[0].secret_key')
+
+# list/confirm contentes of the bucket
+mc tree ${CLUSTER_ID}
+
+# Get ignition_bootstrap token from bucket
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:55]: This step can be skipped.

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  yq r dependencies/openshift4-cloudscale/class/defaults.yml \
+  parameters.openshift4_cloudscale.images.terraform.image)
+tf_tag=$(\
+  yq r dependencies/openshift4-cloudscale/class/defaults.yml \
+  parameters.openshift4_cloudscale.images.terraform.tag)
+  
+# Generate the terraform alias
+alias terraform='docker run -it --rm \
+  -e CLOUDSCALE_TOKEN="${CLOUDSCALE_TOKEN}" \
+  -e TF_VAR_ignition_bootstrap="${TF_VAR_ignition_bootstrap}" \
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:71]:
```suggestion
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+. After all resources are deleted we need to remove the bucket
++
+[source,console]
+----
+# list/confirm contents of the bucket
+mc tree ${CLUSTER_ID}
+
+# delete bootstrap-ignition object
+mc rb "${CLUSTER_ID}/${CLUSTER_ID}-bootstrap-ignition" --force
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:106]: Will this work if there's still files in this bucket?

**Review by srueg — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc (hunk 1 above)
Comment (nunojusto) [reply docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:55]: I agree, it was a step to be sure you are in the correct bucket.
But you will receive that message when you config mc with the host

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc (hunk 3 above)
Comment (nunojusto) [reply docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:106]: Yes, that's why the command is issued with "--force"

**Comment by tobru** [comment]
Missing steps:
* Remove objects user in cloudscale.ch after removing the buckets
* Cleanup entries in Vault
* Remove LDAP whitelisting and service in VSHN portal

Please also try to not duplicate content, there are steps in this page which are the same in the installation page. For this either use Antora partials or refer to the steps in the installation page.

For partials, see https://docs.antora.org/antora/2.3/page/include-a-partial/ and as an example https://github.com/vshn/kb/blob/bb60d25d9cb20176cfafe538b3eaf1ba40ffbab0/docs/modules/help/pages/faq.adoc

**Comment by nunojusto** [comment]
> Missing steps:
> 
> * Remove objects user in cloudscale.ch after removing the buckets
> * Cleanup entries in Vault
> * Remove LDAP whitelisting and service in VSHN portal
> 
> Please also try to not duplicate content, there are steps in this page which are the same in the installation page. For this either use Antora partials or refer to the steps in the installation page.
> 
> For partials, see https://docs.antora.org/antora/2.3/page/include-a-partial/ and as an example https://github.com/vshn/kb/blob/bb60d25d9cb20176cfafe538b3eaf1ba40ffbab0/docs/modules/help/pages/faq.adoc

I think all is corrected. Can you review, please?

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  jq -e ".[] | select(.display_name == \"${CLUSTER_ID}\")")
+
+# configure minio client to use the bucket
+mc config host add \
+  "${CLUSTER_ID}" "https://objects.${REGION}.cloudscale.ch" \
+  $(echo $response | jq -r '.keys[0].access_key') \
+  $(echo $response | jq -r '.keys[0].secret_key')
+
+# delete bootstrap-ignition object
+mc rb "${CLUSTER_ID}/${CLUSTER_ID}-bootstrap-ignition" --force
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:61]: Will this work if there's still content in the bucket?
I think you have to run `mc rm -r --force "${CLUSTER_ID}/${CLUSTER_ID}/"` first

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+# delete image-registry object
+mc rb "${CLUSTER_ID}/${CLUSTER_ID}-image-registry --force
+
+# delete cloudscale.ch user object
+curl -i -H "Authorization: Bearer ${CLOUDSCALE_TOKEN}" -X DELETE $(echo $response | jq -r '.href')
+----
+
+. Delete vault entries (you can find the credentials in the https://password.vshn.net/cred/detail/2162[password manager]):
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:70]: We have login with LDAP now, no need for credentials.

```suggestion
. Delete vault entries:
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
++
+[source,console]
+----
+# Vault login
+export VAULT_ADDR=https://vault.syn.vshn.net
+vault login
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:76]:
```suggestion
vault login -method=ldap username=<your.name>
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+# delete cloudscale.ch user object
+curl -i -H "Authorization: Bearer ${CLOUDSCALE_TOKEN}" -X DELETE $(echo $response | jq -r '.href')
+----
+
+. Delete vault entries (you can find the credentials in the https://password.vshn.net/cred/detail/2162[password manager]):
++
+[source,console]
+----
+# Vault login
+export VAULT_ADDR=https://vault.syn.vshn.net
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:75]:
```suggestion
export VAULT_ADDR=https://vault-prod.syn.vshn.net
```

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+vault login
+
+# delete token secret
+vault kv delete kv/${TENANT_ID}/${CLUSTER_ID}/cloudscale
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:79]: The setup in Vault changed. The secrets are now in `clusters/kv/${TENANT_ID}/${CLUSTER_ID}/`

```suggestion
vault kv delete clusters/kv/${TENANT_ID}/${CLUSTER_ID}/cloudscale
```

**Review by srueg — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc (hunk 4 above)
Comment (nunojusto) [reply docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:61]: For me it worked

File: docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc (hunk 8 above)
Comment (nunojusto) [reply docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:79]: If this is changed, it needs also to change in Install documentation

Comment (srueg) [reply docs/modules/ROOT/pages/how-tos/cloudscale/decommission.adoc:79]: Good point, yes

**Review by tobru — APPROVED** [review]
Fine by me - please also let @corvus-ch have a review before merging.

**Review by srueg — APPROVED** [review]

**Review by corvus-ch — APPROVED** [review]

**Pull request merged**